pallets/sto/src/benchmarking.rs

```diff
@@ -0,0 +1,215 @@
+            total: 1.into(),
+            price: ((i + 1) as u128).into(),
+        })
+    }
+    tiers
+}
+
+fn setup_fundraiser<T: Trait>() -> Result<(User<T>, Ticker), DispatchError> {
+    let alice = <UserBuilder<T>>::default().generate_did().build("alice");
+    let alice_portfolio = PortfolioId::default_portfolio(alice.did());
+
+    let offering_ticker = Ticker::try_from(&[b'A'][..]).unwrap();
+    let raise_ticker = Ticker::try_from(&[b'B'][..]).unwrap();
+    create_asset::<T>(alice.origin(), offering_ticker, 1_000_000)?;
+    create_asset::<T>(alice.origin(), raise_ticker, 1_000_000)?;
+
+    add_compliance::<T>(alice.origin(), offering_ticker)?;
+    add_compliance::<T>(alice.origin(), raise_ticker)?;
+
+    let venue_id = <Settlement<T>>::venue_counter();
+    <Settlement<T>>::create_venue(
+        alice.origin().into(),
+        VenueDetails::default(),
+        vec![alice.account()],
+        VenueType::Sto,
+    )?;
+
+    <Sto<T>>::create_fundraiser(
+        alice.origin().into(),
+        alice_portfolio,
+        offering_ticker,
+        alice_portfolio,
+        raise_ticker,
+        generate_tiers::<T>(1),
+        venue_id,
+        None,
+        None,
+    )?;
+
+    Ok((alice, offering_ticker))
+}
+
+benchmarks! {
+    _ {}
+
+    create_fundraiser {
+        // Number of tiers
+        let i in 1 .. MAX_TIERS as u32;
+
+        let alice = <UserBuilder<T>>::default().generate_did().build("alice");
+        let alice_portfolio = PortfolioId::default_portfolio(alice.did());
+
+        let offering_ticker = Ticker::try_from(&[b'A'][..]).unwrap();
+        let raise_ticker = Ticker::try_from(&[b'B'][..]).unwrap();
+        create_asset::<T>(alice.origin(), offering_ticker, 1_000_000)?;
+        create_asset::<T>(alice.origin(), raise_ticker, 1_000_000)?;
+
+        add_compliance::<T>(alice.origin(), offering_ticker)?;
+        add_compliance::<T>(alice.origin(), raise_ticker)?;
+
+        let venue_id = <Settlement<T>>::venue_counter();
+        <Settlement<T>>::create_venue(
+            alice.origin().into(),
+            VenueDetails::default(),
+            vec![alice.account()],
+            VenueType::Sto
+        )?;
+    }: _(
+            alice.origin(),
+            alice_portfolio,
+            offering_ticker,
+            alice_portfolio,
+            raise_ticker,
+            generate_tiers::<T>(i),
+            venue_id,
+            None,
+            None
+        )
+    verify {
+        ensure!(FundraiserCount::get(offering_ticker) > 0, "create_fundraiser");
+    }
+
+    invest {
+        // Rule complexity
+        let c in 1 .. T::MaxConditionComplexity::get() as u32;
+
+        let alice = <UserBuilder<T>>::default().generate_did().build("alice");
+        let alice_portfolio = PortfolioId::default_portfolio(alice.did());
+        let bob = <UserBuilder<T>>::default().generate_did().build("bob");
+        let bob_portfolio = PortfolioId::default_portfolio(bob.did());
```

Comment:
Occurs 4x when the name is parameterized; let's make a function out of this returning a tuple and taking a string slice.

Reply:
A more general comment about development: I'd recommend eagerly deduplicating code / extracting to functions instead of ^C^V-ing... This helps you find common patterns, possibly uncovering something fundamental and "domain primitives". Moreover, it saves time in reviews, improving the focus on more semantic parts (and if a bug is found, it is sure to be fixed in all places...). For my own part, I basically never ^C^V code.